docs/docs/domain.mdx

```diff
@@ -465,64 +470,180 @@ stories:
 
 ### Slot Auto-fill
 
-The slot will be set automatically if your NLU model picks up an entity, and your domain
- contains a slot with the same name, provided the following conditions are met:
- 
-1. `store_entities_as_slots` is set to true
-2. the slot's `auto_fill` property is set to true
+:::caution Removed in 3.0
```

Comment:
We don't have a clear guideline for this at the moment. The only guideline we have now is that we remove callouts when we remove deprecated features / behaviours in order to start fresh in a new major and reducing the amount of callouts in a doc page (i.e. it doesn't scale otherwise). I would assume the same for this case (even if nothing was deprecated before).

How about:
1. remove this section altogether as it doesn't apply to 3.x
2. in the callout that announces the new feature (i.e. ```:::info New in 3.0```), use some of the text that you have here to briefly explain the behaviour change and link at the migration guide? (like @melindaloubser1  suggested)

?

Reply:
I would add such callout in [the next section below](https://github.com/RasaHQ/rasa/pull/10011/files#diff-cb86102a1484b02650a25eceae48c41c59f43802ab70061e6dd1d120327cd0e6R483-R484), as well as in `docs/docs/forms.mdx`. I think it's good to add duplicated callouts for the same feature wherever we need them (i.e. across different docs pages), because it's more explicit to the user!

🤔 I think I should add this as a documentation guideline, WDYT?

Reply:
Sounds good, thank you 👍🏼